### PR TITLE
Update system to allow associating multiple terms for one result

### PR DESCRIPTION
**Pull request by GermanGolota**

### CODE REVIEW

File: src/Aurora.Application/Entities/SearchRequest.cs [hunk 1]
```diff
@@ -11,11 +11,11 @@ public class SearchRequest
         public Guid Id { get; set; }
 
         [MaxLength(255)]
-        public string SearchTerm { get; set; }
```
Comment (RuslanPr0g) [inline src/Aurora.Application/Entities/SearchRequest.cs:14]: bad practice to put nulls right here, make it at least as a nuallableldfsjgsd type

File: src/Aurora.Application/Entities/SearchRequest.cs [hunk 2]
```diff
@@ -11,11 +11,11 @@ public class SearchRequest
...
+        public string SearchTerm { get; set; } = null!;
         public SupportedWebsite Website { get; set; }
         public ContentType ContentType { get; set; }
 
         public int OccurredCount { get; set; }
-        public IEnumerable<SearchRequestQueueItem> QueueItems { get; set; }
+        public IEnumerable<SearchRequestQueueItem> QueueItems { get; set; } = null!;
```
Comment (RuslanPr0g) [inline src/Aurora.Application/Entities/SearchRequest.cs:19]: the same fopadsg

File: src/Aurora.Application/Entities/SearchRequestToResult.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+{
+    public class SearchRequestToResult
+    {
+        [Key]
+        public Guid Id { get; set; }
+        public Guid SearchRequestId { get; set; }
+        public Guid SearchResultId { get; set; }
+
+        public SearchRequest SearchRequest { get; set; } = null!;
+        public SearchResult SearchResult { get; set; } = null!;
```
Comment (RuslanPr0g) [inline src/Aurora.Application/Entities/SearchRequestToResult.cs:18]: the same tiurwehtiwesajtfadfsgdfsg

File: src/Aurora.Application/Entities/SearchRequestToResult.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+namespace Aurora.Application.Entities
+{
+    public class SearchRequestToResult
+    {
+        [Key]
+        public Guid Id { get; set; }
+        public Guid SearchRequestId { get; set; }
+        public Guid SearchResultId { get; set; }
+
+        public SearchRequest SearchRequest { get; set; } = null!;
```
Comment (RuslanPr0g) [inline src/Aurora.Application/Entities/SearchRequestToResult.cs:17]: abopoba the sdmae

File: src/Aurora.Application/Entities/SearchResult.cs [hunk 5]
```diff
@@ -11,7 +12,6 @@ public class SearchResult
         public string? SearchItemUrl { get; set; }
         public DateTime FoundTimeUtc { get; set; }
 
-        public Guid RequestId { get; set; }
-        public SearchRequest Request { get; set; }
+        public IEnumerable<SearchRequestToResult> Requests { get; set; } = null!;
```
Comment (RuslanPr0g) [inline src/Aurora.Application/Entities/SearchResult.cs:15]: the same is here

File: src/Aurora.Application/Extensions/LoggerExtensions.cs [hunk 6]
```diff
@@ -11,9 +11,9 @@ public static class LoggerExtensions
         public static void LogRequest(this ILogger logger, SearchRequestDto request, string prefix)
         {
             logger.LogInformation(
-            "{prefix} for request for '{term}' term with '{types}' types in '{websites}' websites",
+            "{prefix} for request for '{term}' terms with '{types}' types in '{websites}' websites",
             prefix,
-            request.SearchTerm,
+            request.SearchTerms.CommaSeparate(),
```
Comment (RuslanPr0g) [inline src/Aurora.Application/Extensions/LoggerExtensions.cs:16]: why dont make it generic separation but select separation charactre d???  ??  ? ?

File: src/Aurora.Infrastructure/Migrations/20220807062709_ReworkResultRelation.cs [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+            migrationBuilder.CreateIndex(
+                name: "IX_RequestToResult_SearchRequestId",
+                table: "RequestToResult",
+                column: "SearchRequestId");
+
+            migrationBuilder.CreateIndex(
+                name: "IX_RequestToResult_SearchResultId_SearchRequestId",
+                table: "RequestToResult",
+                columns: new[] { "SearchResultId", "SearchRequestId" },
+                unique: true);
```
Comment (RuslanPr0g) [inline src/Aurora.Infrastructure/Migrations/20220807062709_ReworkResultRelation.cs:58]: should it be uniquqe?

**Review by RuslanPr0g — CHANGES_REQUESTED** [review]

**Review by RuslanPr0g — APPROVED** [review]

File: src/Aurora.Application/Extensions/LoggerExtensions.cs (hunk 6 above)
Comment (GermanGolota) [reply src/Aurora.Application/Extensions/LoggerExtensions.cs:16]: For easier end-user usage

File: src/Aurora.Infrastructure/Migrations/20220807062709_ReworkResultRelation.cs (hunk 7 above)
Comment (GermanGolota) [reply src/Aurora.Infrastructure/Migrations/20220807062709_ReworkResultRelation.cs:58]: yes, there is no reason to have one search request (website, content type and search term) be associated with some result several times

**Pull request merged**